drivers/platform/surface/surface_platform_profile.c

```diff
@@ -14,67 +14,93 @@
 
 #include <linux/surface_aggregator/device.h>
 
-enum ssam_tmp_profile {
-	SSAM_TMP_PROFILE_NORMAL             = 1,
-	SSAM_TMP_PROFILE_BATTERY_SAVER      = 2,
-	SSAM_TMP_PROFILE_BETTER_PERFORMANCE = 3,
-	SSAM_TMP_PROFILE_BEST_PERFORMANCE   = 4,
+// Enum for the platform performance profile sent to the TMP module.
+enum ssam_perf_profile {
```

Comment:
I'm actually wondering now whether we should keep `tmp` in this name as well... so just keep this as it was; `ssam_tmp_profile` and introduce the `ssam_fan_profile`. The device struct would still be the more generic `ssam_platform_profile_device`?

Naming is hard if we don't really know what these things mean or do :(

Reply:
Yeah. I'd vote for `ssam_perf_profile` and `ssam_fan_profile`. I think that best reflects what we know (so far). No need to add the `tmp` everywhere. I think it might be good to keep in direct SAM interface function definitions, but we don't need it for the rest. Especially if we separate into those two categories.

Reply:
On the other hand we could also do `ssam_tmp_profile` and `ssam_fan_profile` and separate them by the target category.

Reply:
Yeah, I think
> On the other hand we could also do ssam_tmp_profile and ssam_fan_profile and name them by the target category.

Is the best to go with for now tbh, especially since we don't have any further information on the function of the tmp profile setting.

Reply:
`ssam_tmp_profile` is back in place everywhere, delta is much smaller now.